pkg/apis/operator/base/common.go

```diff
@@ -291,6 +295,13 @@ type ResourceRequirementsOverride struct {
 	corev1.ResourceRequirements
 }
 
+type EnvRequirementsOverride struct {
+	// The container name
+	Container string `json:"container"`
+	// The desired EnvVarRequirements
+	EnvVars []corev1.EnvVar `json:"envVars,omitempty"`
```

Comment:
I could use `corev1.EnvVar` to make it more redundant as with the resource overrides but I find it more compact to allow group multiple env vars per container name.